src/subscript/__init__.py

```diff
@@ -24,13 +38,43 @@ def getLogger(module_name="subscript"):
     Returns:
         A logger object
     """
-    logging.basicConfig()  # Ensure at least one handler is available
-
     if not module_name:
-        return logging.getLogger("subscript")
+        return getLogger("subscript")
 
     compressed_name = []
     for elem in module_name.split("."):
         if len(compressed_name) == 0 or elem != compressed_name[-1]:
             compressed_name.append(elem)
-    return logging.getLogger(".".join(compressed_name))
+
+    logger = logging.getLogger(".".join(compressed_name))
+
+    formatter = logging.Formatter("%(levelname)s:%(name)s:%(message)s")
+
+    stdout_handler = logging.StreamHandler(sys.stdout)
+    stdout_handler.addFilter(StdoutFilter())
+    stdout_handler.setFormatter(formatter)
+
+    stderr_handler = logging.StreamHandler(sys.stderr)
+    stderr_handler.addFilter(StderrFilter())
+    stderr_handler.setFormatter(formatter)
+
+    logger.addHandler(stdout_handler)
+    logger.addHandler(stderr_handler)
+
+    return logger
+
+
+class StderrFilter(logging.Filter):
+    """Used to filter out log records from ERROR and above"""
+
+    # pylint: disable=too-few-public-methods
+    def filter(self, record):
+        return record.levelno >= logging.ERROR
+
+
+class StdoutFilter(logging.Filter):
+    """Used to filter out log records from WARNING and below"""
+
+    # pylint: disable=too-few-public-methods
+    def filter(self, record):
+        return record.levelno < logging.ERROR
```

Comment:
You could replace these with lambdas.

`stderr_handler.addFilter(lambda record: record.levelno >= logging.ERROR)`